### ISSUE TITLE
IFT and offline reading

### ISSUE THREAD

**Issue by svgeesus** [w3c/IFT#129]
Originally raised during charter review]():

@r12a said:

>  I wondered whether it's in scope to look at how the font transfer technology affects or is handled by offline reading of web pages, including saving pages from the browser. 

Does a local save count as a new origin or is it still attached to the origin from which it is downloaded? That affects what happens if the user traverses a link to another page on the original website while reading the saved version.

**Comment by garretrieger** [comment]
I'm not too familiar with how browsers specifically handle subresources when saving a page. My expectation in this case would be that since in the saved case the page isn't actually being rendered then the browser wouldn't have the context (ie. know which codepoints are needed, for example what if javascript inserts content?) necessary to execute the incremental request. Instead it would make the most sense to just download and save the whole font, which can always be done for a IFT url by just making a regular GET request.

**Comment by r12a** [comment]
Javascript can be run on a saved page that is read offline as long as the .js file was also downloaded (which is standard when you save a page from your browser to your desktop). [And thank goodness for that, because my internet connection has been broken since Sunday and an engineer won't come until Friday, so i'm eeking out my data allowance on my phone by working offline a lot this week.]

But it seems to me that IFT is really only workable when an online connection is available, so perhaps there needs to be something in the spec that says approximately: If a browser saves a web page it will need to save a full copy of the font as well. And if a web page is being read directly from a local drive (which i do a lot of anyway while developing content) then the IFT mechanism should be bypassed.

I'm guessing that there is less of a need for IFT anyway when reading pages from disk, but my concern is to ensure that IFT implementations don't end up _preventing_ full use of the font when no online resource is accessible.  Does that make sense?

**Comment by garretrieger** [comment]
Yes that all makes sense. I'll make some updates to the spec to add that guidance.
